simplified-app-shared/src/main/java/org/nypl/simplified/app/MainSettingsAccountActivity.java

```diff
@@ -890,97 +904,101 @@ protected void onResume() {
     in_account_name_text.setText(MainSettingsAccountActivity.this.account.getName());
     in_account_subtitle_text.setText(MainSettingsAccountActivity.this.account.getSubtitle());
 
+
+    final BooksType userAccount;
+    if (this.account == null) {
+      userAccount = Simplified.getCatalogAppServices().getBooks();
+    } else {
+      userAccount = Simplified.getBooks(this.account, this, Simplified.getCatalogAppServices().getAdobeDRMExecutor());
+    }
+
+    final AccountsDatabaseType accounts_database  = Simplified.getAccountsDatabase(this.account, this);
+    if (!accounts_database.accountGetCredentials().isSome()) {
+      this.sync_table_row.setVisibility(View.GONE);
+      LOG.debug("No user currently signed in, bypassing UI update and Sync Status Initiation.");
+      return;
+    }
+
+    final AccountCredentials creds = ((Some<AccountCredentials>) accounts_database.accountGetCredentials()).get();
+
+    if (syncButtonShouldBeVisible()) {
+      if (this.annotationsManager == null) {
+        this.annotationsManager = new AnnotationsManager(this.account, creds, this);
+      }
+      this.sync_table_row.setVisibility(View.VISIBLE);
+      checkServerSyncPermission(userAccount);
+    } else {
+      this.sync_table_row.setVisibility(View.GONE);
+    }
+
+    in_table_with_code.setVisibility(View.VISIBLE);
+    in_table_signup.setVisibility(View.GONE);
+
+    in_barcode_text.setText(creds.getBarcode().toString());
+    in_barcode_text.setContentDescription(creds.getBarcode().toString().replaceAll(".(?=.)", "$0,"));
+    in_pin_text.setText(creds.getPin().toString());
+    in_pin_text.setContentDescription(creds.getPin().toString().replaceAll(".(?=.)", "$0,"));
```

Comment:
what is this looking for explicitly? can we sanitize the string in another way?

Reply:
This is doing specific work for accessibilty which I'm not sure of. Must have been lumped into this PR from some re-indenting on my part.